### PR TITLE
Added note about nec protocol fix

### PR DESCRIPTION
**Pull request by dudanov**
## Description:

Added note about nec protocol fix.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2534

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/remote_transmitter.rst [hunk 1]
```diff
@@ -183,6 +183,12 @@ Configuration variables:
 
 This :ref:`action <config-action>` sends an NEC infrared remote code to a remote transmitter.
 
+.. note::
+
+    In version 2021.11, the order of transferring bits was corrected from MSB to LSB in accordance with the NEC standard.
+    Therefore, in earlier versions of the configuration, it is necessary to reverse the order of the address and command bits.
```
Comment (davet2001) [inline components/remote_transmitter.rst]: Suggested wording:
Therefore, if the the configuration file has come from an earlier version of ESPhome, it is necessary to reverse the order of the address and command bits when moving to 2021.11 or above.

**Review by davet2001 — CHANGES_REQUESTED** [review]
Minor rewording suggested

**Comment by dudanov** [comment]
> Minor rewording suggested

Done.

**Review by davet2001 — APPROVED** [review]
Looks good! Thanks! 👍

File: components/remote_transmitter.rst [hunk 2]
```diff
@@ -183,6 +183,12 @@ Configuration variables:
 
 This :ref:`action <config-action>` sends an NEC infrared remote code to a remote transmitter.
 
+.. note::
+
+    In version 2021.11, the order of transferring bits was corrected from MSB to LSB in accordance with the NEC standard.
```
Comment (jesserockz) [inline components/remote_transmitter.rst]:
```suggestion
    In version 2021.12, the order of transferring bits was corrected from MSB to LSB in accordance with the NEC standard.
```

File: components/remote_transmitter.rst [hunk 3]
```diff
@@ -183,6 +183,12 @@ Configuration variables:
...
+    Therefore, if the the configuration file has come from an earlier version of ESPhome, it is necessary to reverse the order of the address and command bits when moving to 2021.11 or above.
```
Comment (jesserockz) [inline components/remote_transmitter.rst]:
```suggestion
    Therefore, if the the configuration file has come from an earlier version of ESPhome, it is necessary to reverse the order of the address and command bits when moving to 2021.12 or above.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
This breaking change will not be merged for the .11 release. Will make sure to put it for next though.

**Comment by kbx81** [comment]
Should this note be added to to the remote receiver doc, too? I just discovered this change tonight when my NEC receiver stopped working. 🙃

**Comment by dudanov** [comment]
> Should this note be added to to the remote receiver doc, too? I just discovered this change tonight when my NEC receiver stopped working. upside_down_face

Thank you for reminding about receiver.

**Comment by davet2001** [comment]
> Should this note be added to to the remote receiver doc, too? I just discovered this change tonight when my NEC receiver stopped working. 🙃

Yes! +1 from me.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**